app/controllers/katello/api/v2/repository_sets_controller.rb

```diff
@@ -52,11 +52,21 @@ def available_repositories
         end
       end
 
+      sorted_groups = repos.sort_by do |repo|
+        major, minor = repo[:substitutions][:releasever].nil? ? 1000.1000 : repo[:substitutions][:releasever].split('.').map(&:to_i)
+        major = major == 0 ? 1000 : major
+        minor = minor.nil? ? 1000 : minor
+        repo[:substitutions][:basearch] = repo[:substitutions][:basearch].nil? ? "" : repo[:substitutions][:basearch]
```

Comment:
In this line you're setting basearch to an empty string where it would have been nil in the response. That's probably not what you wanted to do! Let's change this to preserve (not modify) the basearch on the repo itself.